### PR TITLE
[#64] Apps recirculation module

### PR DESCRIPTION
**Pull request by pylapp**
# User story

https://github.com/Orange-OpenSource/ods-ios/issues/64

# Definition of Done

- [x] DTO / BO
- [x] Repository / services
- [x] Unit tests
- [x] Cache for offline mode (check HTTP headers if online, file in offline or errors)
- [x] List of apps: other apps
- [x] List of apps: categories / sections ("list header feature")
- [x] Preview for bigger texts (_peak and pop_)
- [x] App Store redirections
- [x] In configuration, flatten or not the apps (i.e. keep categories or display all)
- [x] About module demo
- [x] Update CHANGELOG
- [x] Update README
- [x] Update docs
- [x] Haptic / vibrations for loading states

# About tests

- [x] Use JSON mock
- [x] iPhone and iPad tests (with screenshots)

# About a11y

- [x] Check a11Y vocalization
- [x] Check dynamic fonts
- [x] Check light and dark modes
- [x] Check clickable area
- [x] Add haptics interactions

### CODE REVIEW

**Comment by pylapp** [comment]
ℹ️ Refactored a bit some sources (typo, etc) and add unit tests for priority rules about module about items

**Comment by pylapp** [comment]
No details about "featured apps" (cf https://github.com/Orange-OpenSource/ods-ios/issues/64#issuecomment-1834644360)

**Comment by pylapp** [comment]
Investigated also the [git-secret](https://github.com/sobolevn/git-secret) project so as to store any configuration file with API keys or other secrets ; but it needs all developers be used to use GPG keys to read and write in encyphered files, so this step cannot be considered today :(

**Comment by pylapp** [comment]
Find attached a ZIP archive with screenshots showing several cases:

- on iPhone: screens in light and dark modes, with categorized or not apps, peak and pop, bigger text sizes, light and dark modes, case without cache of apps icons
- on iPad: peak and pop, categorized or not apps

[#64 - screenshots (iPhone, iPad, a11y, dark and light modes).zip](https://github.com/Orange-OpenSource/ods-ios/files/13954301/64.-.screenshots.iPhone.iPad.a11y.dark.and.light.modes.zip)

File: OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/Configuration/Items/MoreAPps/ODSMoreAppsItem.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
+//
+// Software Name: Orange Design System (iOS)
+// SPDX-FileCopyrightText: Copyright (c) 2021 - 2023 Orange SA
```
[the remark this answers is not in the document]
Comment (pylapp) [reply OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/Configuration/Items/MoreAPps/ODSMoreAppsItem.swift:3]: Ce sera réglé dans une PR ultérieure car un changement aujourd'hui dans les headers sera écrasé par _swiftformat_ en _build phase_.

File: OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/Internal/ListItems/MoreApps/AppsPlusDTO.swift [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+        }
+
+        children = decodedChildren
+    }
+}
+
+// swiftlint:disable force_cast
+extension AppsPlusListDTO {
+
+    var apps: [AppsPlusAppDetailsDTO] {
```
[the remark this answers is not in the document]
Comment (pylapp) [reply OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/Internal/ListItems/MoreApps/AppsPlusDTO.swift:71]: Utilser _compactMap_ plutôt.

File: OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/Internal/ListItems/MoreApps/AppsPlusDTO.swift [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+        children.filter { $0 is AppsPlusAppDetailsDTO }.map { $0 as! AppsPlusAppDetailsDTO }
+    }
+
+    var sections: [AppsPlusSectionDTO] {
```
[the remark this answers is not in the document]
Comment (pylapp) [reply OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/Internal/ListItems/MoreApps/AppsPlusDTO.swift:75]: Utiliser _compactMap_ plutôt.

**Review by pylapp — COMMENTED** [review]
@ludovic35 I processed your comments, and improved the feature :)
Feel free to have a last look and merge!

File: OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/Internal/ListItems/MoreApps/AppsPlusDTO.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (ludovic35) [reply OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/Internal/ListItems/MoreApps/AppsPlusDTO.swift:74]: yes

**Pull request merged**